Cargo.toml

```diff
@@ -30,7 +30,7 @@ rustdoc-args = ["--cfg", "docsrs"]
 fugit = "0.3.5"
 embedded-hal = { version = "0.2.6", features = ["unproven"] }
 embedded-dma = "0.2.0"
-cortex-m = "^0.7.4"
+cortex-m = { version = "^0.7.7", features = ["critical-section-single-core"] }
```

Comment:
Should the feature only be enabled on the single core variants?

Reply:
I might be in over my head here...

From what I've managed to figure out is:
  - `cortex_m::interrupt::free` is [unsound(?)](https://github.com/rust-embedded/cortex-m/pull/448#issuecomment-1211848911)
  - `critical_section::with` is not unsound but needs to be implemented by either this crate or use the "default" provided by the critical-section-single-core feature in cortex-m

https://github.com/rust-embedded/cortex-m/pull/448

https://github.com/rust-embedded/cortex-m/blob/master/CHANGELOG.md states that
- Added critical-section-single-core feature which provides an implementation for the critical_section crate for single-core systems, based on disabling all interrupts. (#447)
- interrupt::free no longer hands out a CriticalSection token because it is unsound on multi-core. Use critical_section::with instead. (#447)

So to answer your question: Probably, in the future when this crate supports multiple cores. As long as we don't provide any way to run code on multiple cores in this crate, using the feature `critical-section-single-core` should be fine(?).

Reply:
Thanks for investigating this @olback ! I'm ok with this solution for now since there's no explicit support for multiple cores yet. It's even possible a HAL isn't actually the right place to implement multicore support, but I think that is still uncertain.